realAaronWu/pravega
# Cherry pick issue 6637

**Change log description**  
Cherry pick https://github.com/pravega/pravega/pull/6527 and https://github.com/pravega/pravega/pull/6628 to r0.10.
**Purpose of the change**  
Fixes https://github.com/pravega/pravega/issues/6635.

**What the code does**  
See https://github.com/pravega/pravega/pull/6527 and https://github.com/pravega/pravega/pull/6628.

**How to verify it**  
See https://github.com/pravega/pravega/pull/6527 and https://github.com/pravega/pravega/pull/6628.
